### DISCUSSION TITLE
inbound & outbound channel question

### DISCUSSION THREAD

**Discussion by YGwan** [spring-projects/spring-integration#9692] [Q&A]
So, even if there are no restrictions on IP address, can the problem occur even if the same inbound and outbound clients are used?
Do you prefer to separate clients or not?

If you have a shared client that handles both inbound and outbound at the same time, do you need to separate the message channel that handles inbound and outbound? It doesn't happen when you do, but if you send a message in a row, you get the following error.

```
2024-12-01T01:25:13.360+09:00  INFO 77625 --- [6c-6705b4e0d855] o.s.i.dispatcher.UnicastingDispatcher    : An exception was thrown by 'bean 'mqttOutboundHandler.serviceActivator.handler' for component 'mqttOutboundHandler.serviceActivator'' while handling 'GenericMessage [payload={
  "message": "Hello AWS IoT TEST"
}, headers={mqtt_receivedRetained=false, mqtt_id=0, mqtt_duplicate=false, id=d5426437-88d4-6a07-5c18-77877ccaedf4, mqtt_receivedTopic=ygwan/test, mqtt_receivedQos=0, timestamp=1732983913359}]': error occurred in message handler [bean 'mqttOutboundHandler'; defined in: 'class path resource [com/mqttService/config/MQTTConfig2.class]'; from source: 'com.conportlab.buildservice.config.MQTTConfig2.mqttOutboundHandler()']. Failing over to the next subscriber.
{
  "message": "Hello AWS IoT TEST"
}
```

This error condition is the error that occurred when inbound and outbound clients took the same message channel and used one message channel, subscribed to the topic "ygwan/test" and received the second message when another device issued the message in that topic twice in a row.

**Comment by artembilan** [comment]
Closed as duplication of: https://github.com/spring-projects/spring-integration/discussions/9686
